specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2024-04-01-preview/fluxconfiguration.json

```diff
@@ -0,0 +1,1822 @@
+          "description": "Base64-encoded certificate used to authenticate a Service Principal ",
+          "type": "string",
+          "x-nullable": true,
+          "x-ms-secret": true
+        },
+        "clientCertificatePassword": {
+          "description": "The password for the certificate used to authenticate a Service Principal ",
+          "type": "string",
+          "x-nullable": true,
+          "x-ms-secret": true
+        },
+        "clientCertificateSendChain": {
+          "description": "Specifies whether to include x5c header in client claims when acquiring a token to enable subject name / issuer based authentication for the Client Certificate",
+          "type": "boolean"
+        }
+      }
+    },
+    "ManagedIdentityDefinition": {
+      "description": "Parameters to authenticate using a Managed Identity.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "clientId": {
+          "description": "The client Id for authenticating a Managed Identity.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "ManagedIdentityPatchDefinition": {
+      "description": "Parameters to authenticate using a Managed Identity.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "clientId": {
+          "description": "The client Id for authenticating a Managed Identity.",
+          "type": "string",
+          "x-nullable": true
+        }
+      }
+    },
+    "OCIRepositoryDefinition": {
+      "description": "Parameters to reconcile to the OCIRepository source kind type.",
+      "type": "object",
+      "x-nullable": true,
+      "properties": {
+        "url": {
+          "description": "The URL to sync for the flux configuration OCI repository.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "timeoutInSeconds": {
+          "description": "The maximum time to attempt to reconcile the cluster OCI repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "syncIntervalInSeconds": {
+          "description": "The interval at which to re-reconcile the cluster OCI repository source with the remote.",
+          "type": "integer",
+          "format": "int64",
+          "default": 600,
+          "x-nullable": true
+        },
+        "repositoryRef": {
+          "description": "The source reference for the OCIRepository object.",
+          "$ref": "#/definitions/OCIRepositoryRefDefinition"
+        },
+        "layerSelector": {
+          "description": "The layer to be pulled from the OCI artifact.",
+          "$ref": "#/definitions/LayerSelectorDefinition"
+        },
+        "verify": {
+          "description": "Verification of the authenticity of an OCI Artifact.",
+          "$ref": "#/definitions/VerifyDefinition"
+        },
+        "insecure": {
```

Comment:
>insecure [](http://example.com/codeflow?start=9&length=8)

Recommend enum instead of boolean. Consider something like connectionSecurity: { TLS, None } or similar.

Reply:
We are integrating open source fields. We need to keep it consistent to avoid confusion with the customers. [https://fluxcd.io/flux/components/source/ocirepositories/#insecure](https://fluxcd.io/flux/components/source/ocirepositories/#insecure)

Also, some other sources(gitRepository and bucket) part of this json have been using insecure and changing this would make it inconsistent.